src/main/kotlin/org/neo4j/graphql/handler/filter/OptimizedFilterHandler.kt

```diff
@@ -0,0 +1,369 @@
+package org.neo4j.graphql.handler.filter
```

Comment:
pretty large file with pretty large functions, can we split this up a bit more to make it more understandable?